iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs

```diff
@@ -92,7 +91,7 @@ private async Task InitializeAndSetupClientAsync(CancellationToken cancellationT
             if (ShouldClientBeInitialized(s_connectionStatus))
             {
                 // Allow a single thread to dispose and initialize the client instance.
-                await _initSemaphore.WaitAsync();
+                await _initSemaphore.WaitAsync(cancellationToken);
```

Comment:
May need to handle an exception that is thrown if this is cancelled.

Reply:
Same with the calls below.

Reply:
I think it's best if catch that exception at the program entry and log it. let me what you think.